test/test_bbm_1d.jl

```diff
@@ -41,6 +41,23 @@ end
     end
 
     @test_allocations(semi, sol, allocs=5_000)
+
+    # test PeriodicRationalDerivativeOperator
+    D1 = periodic_derivative_operator(1, accuracy_order, xmin(mesh), xmax(mesh),
+                                      nnodes(mesh))
+    D2 = D1^2
+    solver = Solver(D1, D2)
```

Comment:
If we go as far as this, it could make sense to add downstream tests to SummationByPartsOperators.jl. I am not sure whether I have documented that `D1^` will necessarily return a `PeriodicRationalDerivativeOperator`. It should be pretty stable, but we use some parts that are less clearly described as public API.

Reply:
The same also holds for `FourierPolynomialDerivativeOperator`, doesn't it? As far as I can tell, it's also not clearly documented that `D1^` will return a `FourierPolynomialDerivativeOperator` if `D1` is a `FourierDerivativeOperator`.

Reply:
Well, I have not added any documentation to it, so it is indeed not really public but more like an implementation detail to make `^` etc. work - which is documented

> All of these periodic derivative operators support multiplication and addition such that polynomials and rational functions of them can be represented efficiently

already in the README.md (see https://github.com/ranocha/SummationByPartsOperators.jl?tab=readme-ov-file#periodic-domains). 

But we can go ahead and merge this PR here. We don't need to add downstream tests right now, but it may be nice to keep this in mind for the future.

Reply:
The simpler option would be to add some explicit tests that `^` returns the appropriate types with some comments that DispersiveShallowWater.jl uses this interface.

Reply:
> The same also holds for `FourierPolynomialDerivativeOperator`, doesn't it? As far as I can tell, it's also not clearly documented that `D1^` will return a `FourierPolynomialDerivativeOperator` if `D1` is a `FourierDerivativeOperator`.

Yes